cloud/terraform/templates.go

```diff
@@ -265,6 +265,7 @@ CONTROL_PLANE_VERSION={{ .Machine.Spec.Versions.ControlPlane }}
 CLUSTER_DNS_DOMAIN={{ .Cluster.Spec.ClusterNetwork.ServiceDomain }}
 POD_CIDR={{ getSubnet .Cluster.Spec.ClusterNetwork.Pods }}
 SERVICE_CIDR={{ getSubnet .Cluster.Spec.ClusterNetwork.Services }}
+CNI_PROVIDER=weavenet
```

Comment:
We want to use Calico no?

Reply:
If not, and if we want to support multiple CNI providers, then maybe we should talk about exposing that as a config setting in Cluster type.

Reply:
I will change it to calico. We may want to provide the ability to switch between providers. But it need changes at  ClusterSpec/ClusterNetworkingConfig(https://github.com/cadmuxe/cluster-api/blob/master/pkg/apis/cluster/v1alpha1/cluster_types.go). Have discussed this with @satyasm , we can update spec interface later when we have more context.

Reply:
If we do really need to support multiple CNI providers in the API itself, we should get the ball rolling on that now. That could be an alpha milestone.

> We may want to provide the ability to switch between providers.

I have another follow up question - would the machine controller's business logic need to change at all depending on what CNI we install here?

Reply:
IMO the configuration needs to co-ordinate a fair amount of things. For example, for the default we are co-ordination calico, with usePodCIDR mode and the RangeAllocator on the cluster. This PR just attempts to make this configuration the default instead of the previous weave based config. I agree that we should address the configurable CNI problem but that sounds like a PR in itself that considers all the factors, right?

Reply:
Correct. So then should we simply swap the CNI from weave to calico (I'm okay with that). If we want to support multiple CNI providers -- what this PR does, we should have a formal proposal IMO.